src/pytest_plugins/filler/tests/test_solc.py

```diff
@@ -0,0 +1,201 @@
+
+
+@pytest.fixture(scope="module")
+def create_clean_solc_select_environment(request):
+    """
+    Setup: Copies solc artifacts to a temporary location before starting tests
+    Teardown: Restores the artifacts after tests are done.
+    """
+    try:
+        test_session_solc_version, _ = solc_select.solc_select.current_version()
+    except Exception as e:
+        raise Exception(
+            "Error in setup: ensure you've called `solc-select use <version>` before running the "
+            f"framework tests (exception: {e})"
+        )
+
+    artifacts_dir = solc_select.constants.ARTIFACTS_DIR
+    global_version_file = solc_select.constants.SOLC_SELECT_DIR.joinpath("global-version")
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        temp_dir_path = Path(temp_dir)
+
+        # Copy global-version and artifacts to a temporary directory and delete them
+        if global_version_file.exists():
+            shutil.copy(global_version_file, temp_dir_path / "global-version")
+            os.remove(global_version_file)
+        if artifacts_dir.exists():
+            shutil.copytree(artifacts_dir, temp_dir_path / "artifacts", dirs_exist_ok=True)
+            shutil.rmtree(artifacts_dir)
+
+        os.makedirs(artifacts_dir, exist_ok=True)  # this won't get recreated by solc-select
+
+        yield
+        print("Running teardown")
+        # Teardown: Restore the original files and directory from the temporary location
+        if global_version_file.exists():
+            os.remove(global_version_file)
+        if artifacts_dir.exists():
+            shutil.rmtree(artifacts_dir)
+
+        # Restore the global-version file and artifacts
+        if (temp_dir_path / "global-version").exists():
+            shutil.copy(temp_dir_path / "global-version", global_version_file)
+        if (temp_dir_path / "artifacts").exists():
+            shutil.copytree(temp_dir_path / "artifacts", artifacts_dir, dirs_exist_ok=True)
+
+        # Restore the solc version
+        solc_select.solc_select.switch_global_version(
+            str(test_session_solc_version), always_install=True
+        )
+
+
+@pytest.mark.usefixtures("create_clean_solc_select_environment")
+@pytest.mark.parametrize("solc_version", ["0.8.21", "0.8.26"])
+class TestSolcVersion:  # noqa: D101
+    def test_solc_versions_flag(self, pytester, solc_version):
+        """
+        Ensure that the version specified by the `--solc-version` gets installed and is used.
+        """
+        pytester.makeconftest(
+            f"""
+            import pytest
+            from ethereum_test_tools.code import Solc
+
+            @pytest.fixture(autouse=True)
+            def check_solc_version(request):
+                assert request.config.getoption("solc_version") == "{solc_version}"
+                #solc_bin_version = Solc(request.config.getoption("solc_bin")).version
+                #assert request.config.getoption("solc_version") == solc_bin_version
```

Comment:
```suggestion
```
Should these be removed?

Reply:
Thanks for catching this! I've added back a check that the `Solc` instance we create from the `solc_bin` fixture reports the expected version. This is the actual test :slightly_smiling_face:. cf c3b5f2a58.